[BUG] InternalEngineTests.testForceMergeWithSoftDeletesRetentionAndRecoverySource failure

This is reproducible:

```
./gradlew ':server:test' --tests "org.opensearch.index.engine.InternalEngineTests.testForceMergeWithSoftDeletesRetentionAndRecoverySource" -Dtests.seed=1766E6A3B733A7AA -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=lv-LV -Dtests.timezone=Asia/Kathmandu -Druntime.java=19

org.opensearch.index.engine.InternalEngineTests > testForceMergeWithSoftDeletesRetentionAndRecoverySource FAILED
    java.lang.AssertionError: 
    Expected: a collection with size <5>
         but: collection size was <6>
        at __randomizedtesting.SeedInfo.seed([1766E6A3B733A7AA:9169E6934A38144D]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:964)
        at org.junit.Assert.assertThat(Assert.java:930)
        at org.opensearch.index.engine.InternalEngineTests.testForceMergeWithSoftDeletesRetentionAndRecoverySource(InternalEngineTests.java:1907)
```

https://build.ci.opensearch.org/job/gradle-check/7223/consoleFull

Another instance here: https://build.ci.opensearch.org/job/gradle-check/10777

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.index.engine.InternalEngineTests.testForceMergeWithSoftDeletesRetentionAndRecoverySource" -Dtests.seed=C75D09DC5A0C3458 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ms -Dtests.timezone=Europe/Bucharest -Druntime.java=19

org.opensearch.index.engine.InternalEngineTests > testForceMergeWithSoftDeletesRetentionAndRecoverySource FAILED
    java.lang.AssertionError: 
    Expected: a collection with size <3>
         but: collection size was <5>
        at __randomizedtesting.SeedInfo.seed([C75D09DC5A0C3458:415209ECA70787BF]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:964)
        at org.junit.Assert.assertThat(Assert.java:930)
        at org.opensearch.index.engine.InternalEngineTests.testForceMergeWithSoftDeletesRetentionAndRecoverySource(InternalEngineTests.java:1908)
```

--- a/server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java
+++ b/server/src/test/java/org/opensearch/index/engine/InternalEngineTests.java
@@ -1872,7 +1872,7 @@ public class InternalEngineTests extends EngineTestCase {
         try (
             Store store = createStore();
             InternalEngine engine = createEngine(
-                config(indexSettings, store, createTempDir(), newMergePolicy(), null, null, globalCheckpoint::get)
+                config(indexSettings, store, createTempDir(), newMergePolicy(random(), false), null, null, globalCheckpoint::get)
             )
         ) {
             int numDocs = scaledRandomIntBetween(10, 100);

